### DISCUSSION TITLE
Ear Trumpet will not switch program's volume when changing the active device

### DISCUSSION THREAD

**Discussion by suctionDunk** [File-New-Project/EarTrumpet#742] [Q&A] [answered]
When I right-click the icon and choose a new device,
![image](https://user-images.githubusercontent.com/77592246/104858152-439f3c00-58eb-11eb-8dc0-89016390b334.png)

NONE of my programs switch over. I now have to switch each program individually. I have tried resetting the default playback and comm devices and nothing changed. Several full restarts also did nothing. Anyone know the reason for this?

**Comment by 0lm** [comment]
Im using EarTrumpe 2.1.7.0 and for me everything works properly. Switching between Main Devices, Audios also switch. (except a few unique certain Programs which wont play a sound on the new device when switching, but this is an issue with these certain softwares/games, not with Eartrumpet).
So, this is probably an issue only you (or only a few people?) have. Would be hard to reproduce then with the current information you gave.
But then again, I use the Chocolatey portable version instead of the Microsoft one. Download the chocolatey version and open it with 7zip. Go to the "tools" folder and excract the "Release.zip" inside of it. In "release.zip", its complete content is basically the compiled portable application of eartrumpet 2.1.7.0. Maybe This one will work better than the Microsoft one.

**Comment by riverar** [comment, marked as answer]
Hi @suctionDunk, I converted your issue into a discussion. Changing the default playback device via the context menu (per your screenshot) does not automatically redirect app output devices. Some apps may require a restart. You can also try to redirect their audio via EarTrumpet: Right-click the app, click the double arrows, and choose a new device (example pictured below).

![image](https://user-images.githubusercontent.com/475132/107896683-d253a880-6eeb-11eb-870c-66164c4dc805.png)
